lib/msal-core/src/PopTokenGenerator.ts

```diff
@@ -0,0 +1,153 @@
+import { BrowserCrypto } from './BrowserCrypto';
+import { Base64 } from 'js-base64';
+import { HashAlgorithm, getStringFromArrayBuffer, KeyFormat, utf8Encode, getArrayBufferFromString, KeyUsages, KeyGenAlgorithm, getJwkString } from './PopTokenCommon';
+import { Utils } from './Utils';
+import { DatabaseCache } from './DatabaseCache';
+import { AuthenticationParameters } from './AuthenticationParameters';
+
+type CachedKeyPair = {
+    publicKey: CryptoKey,
+    privateKey: CryptoKey,
+    requestMethod: string,
+    requestUri: string
+}
+
+export class PopTokenGenerator {
+    private _crypto: BrowserCrypto;
```

Comment:
do we have a naming convention one way or the other with _underscore?

Reply:
I believe we actually don't want to do underscores, I'll clean this up.